src/main/java/com/commercetools/sync/categories/impl/CategorySyncImpl.java

```diff
@@ -0,0 +1,129 @@
+
+public class CategorySyncImpl implements CategorySync {
+    private int updatedCategories;
+    private int createdCategories;
+    private int failedCategories;
+    private int processedCategories;
+    private final Logger LOGGER = LoggerFactory.getLogger(CategorySyncImpl.class);
+
+    private CategorySyncOptions options;
+    private TypeService typeService;
+    private CategoryService categoryService;
+
+    public CategorySyncImpl(CategorySyncOptions options, TypeService typeService, CategoryService categoryService) {
+        this.options = options;
+        this.updatedCategories = 0;
+        this.createdCategories = 0;
+        this.failedCategories = 0;
+        this.processedCategories = 0;
+        this.typeService = typeService;
+        this.categoryService = categoryService;
+    }
+
+    @Override
+    public void syncCategories(@Nonnull List<Category> categories) {
+    }
+
+    // TODO: REFACTOR
+    @Override
+    public void syncCategoryDrafts(@Nonnull List<CategoryDraft> categoryDrafts) {
+        processedCategories = categoryDrafts.size();
+        LOGGER.info(format("About to sync %d category drafts into CTP project with key '%s'."
+                , categoryDrafts.size(), options.getCtpProjectKey()));
+        final SyncResult<Category> syncResult = SyncResult.emptyResult();
+
+
+        for (int i = 0; i < categoryDrafts.size(); i++) { // TODO: HANDLE IF THERE IS A NULL OBJECT IN THE LIST!
+            SyncResult<Category> currentSyncResult = SyncResult.emptyResult();
+            CategoryDraft newCategoryDraft = categoryDrafts.get(i);
+            String externalId = newCategoryDraft != null ? newCategoryDraft.getExternalId() : null;
+            if (externalId != null) { // TODO NEED TO PARALLELISE!
+                Category oldCategory = categoryService.fetchCategoryByExternalId(externalId);
+                if (oldCategory == null) {
+                    currentSyncResult = createCategory(newCategoryDraft);
+                } else {
+                    currentSyncResult = CategorySyncUtils.buildActions(oldCategory, newCategoryDraft, typeService);
+                    final List<UpdateAction<Category>> updateActions = currentSyncResult.getUpdateActions();
+                    if (!updateActions.isEmpty()) {
+                        updateCategory(oldCategory, currentSyncResult);
+                    }
+                }
+            }
+            syncResult.mergeStatistics(currentSyncResult);
+        }
+        LOGGER.info(getSummary(syncResult));
+    }
+
+    @Nullable
+    private SyncResult<Category> createCategory(@Nonnull final CategoryDraft newCategory) {
+        SyncResult<Category> syncResult = SyncResult.emptyResult();
+        try {
+            categoryService.createCategory(newCategory);
+            createdCategories++;
+        } catch (Exception e) {
+            syncResult = SyncResult.ofError(SyncError.of(null, format("Failed to create category with external id" +
```

Comment:
If error is important i would log it now and not wait unit my application has crashed and nothing is logged.